aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -315,4 +338,261 @@ CUDAGraph::~CUDAGraph() {
+          }
+        }
+        param_index++;
+      }
+
+    } else if (type == cudaGraphNodeTypeMemcpy) {
+      cudaMemcpy3DParms memcpyParams1;
+      AT_CUDA_CHECK(cudaGraphMemcpyNodeGetParams(node, &memcpyParams1));
+      auto srcPtrResult = checkAllocationWithinGraph(memcpyParams1.srcPtr.ptr, allocations_);
+      auto dstPtrResult = checkAllocationWithinGraph(memcpyParams1.dstPtr.ptr, allocations_);
+      if (!srcPtrResult && !dstPtrResult) {
+        continue;
+      }
+      
+      graph_node_param_updates_.push_back({
+        .node = node,
+        .compute_new_params = [memcpyParams1, srcPtrResult, dstPtrResult](std::vector<void*> actualDataPtrs) {
+          cudaPitchedPtr srcPtr = memcpyParams1.srcPtr;
+          cudaPitchedPtr dstPtr = memcpyParams1.dstPtr;
+          if (srcPtrResult) {
+            auto [alloc_idx, offset] = *srcPtrResult;
+            srcPtr.ptr = (char*)actualDataPtrs[alloc_idx] + offset;
+          }
+          if (dstPtrResult) {
+            auto [alloc_idx, offset] = *dstPtrResult;
+            dstPtr.ptr = (char*)actualDataPtrs[alloc_idx] + offset;
+          }
+          return cudaGraphNodeParams{
+            .type = cudaGraphNodeTypeMemcpy,
+            .memcpy = {
+              .copyParams = {
+                .srcArray = memcpyParams1.srcArray,
+                .srcPos = memcpyParams1.srcPos,
+                .srcPtr = srcPtr,
+                .dstArray = memcpyParams1.dstArray,
+                .dstPos = memcpyParams1.dstPos,
+                .dstPtr = dstPtr,
+                .extent = memcpyParams1.extent,
+                .kind = memcpyParams1.kind
+              },
+            },
+          };
+        },
+      });
+    } else if (type == cudaGraphNodeTypeMemset) {
+      cudaMemsetParams memsetParams1;
+      AT_CUDA_CHECK(cudaGraphMemsetNodeGetParams(node, &memsetParams1));
+
+      auto dstPtrResult = checkAllocationWithinGraph(memsetParams1.dst, allocations_);
+      if (!dstPtrResult) {
+        continue;
+      }
+      
+      graph_node_param_updates_.push_back({
+        .node = node,
+        .compute_new_params = [memsetParams1, dstPtrResult](std::vector<void*> actualDataPtrs) {
+          void* dstPtr = memsetParams1.dst;
+          if (dstPtrResult) {
+            auto [alloc_idx, offset] = *dstPtrResult;
+            dstPtr = (char*)actualDataPtrs[alloc_idx] + offset;
+          }
+          return cudaGraphNodeParams{
+            .type = cudaGraphNodeTypeMemset,
+            .memset = {
+              .dst = dstPtr,
+              .pitch = memsetParams1.pitch,
+              .value = memsetParams1.value,
+              .elementSize = memsetParams1.elementSize,
+              .width = memsetParams1.width,
+              .height = memsetParams1.height,
+            },
+          };
+        },
+      });
+    } else if (type == cudaGraphNodeTypeGraph || type == cudaGraphNodeTypeConditional) {
+      throw std::runtime_error("horrible. please don't make me support these.");
```

Comment:
Doable to support these. But pytorch is unlikely to use cudaGraphNodeTypeGraph and cudaGraphNodeTypeConditional isn't merged into pytorch yet, so don't worry about it.

Reply:
Yeah. It would be possible, especially now that there is only one capture. So this error message is now a bit too dramatic!